### DISCUSSION TITLE
How can I upload this project on my computer and convert it from Gregorian to Solar?

### DISCUSSION THREAD

**Discussion by EbrahimHamzeh** [vkurko/calendar#59] [Q&A] [answered]
I am fluent in JavaScript programming. I wanted to schedule this project, but I don't know how to run it on my computer and where to start in the code so that we can schedule it. Thank you, please help me

**Comment by vkurko** [comment, marked as answer]
Hi @EbrahimHamzeh 

The calendar is written in [Svelte](https://svelte.dev/), and if you are not familiar with it, then you need to familiarize yourself with this framework first.

After you download the code to your computer, you need to install the dependencies

```bash
npm install
```

then create web root directory and prepare assets

```bash
cp -r docs public
ln -s ../packages/build public/build
```

then edit `public/index.html` and replace the lines

```html
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@event-calendar/build@0.11.2/event-calendar.min.css">
<script src="https://cdn.jsdelivr.net/npm/@event-calendar/build@0.11.2/event-calendar.min.js"></script>
```

with

```html
<link rel="stylesheet" href="build/event-calendar.min.css">
<script src="build/event-calendar.min.js"></script>
```

then start the dev server

```bash
npm run dev
```
and access it at http://localhost:8080/.

The project is divided into packages. There is a `core` package, as well as packages responsible for displaying the calendar in the month, week, etc. modes. Examine the source code and make your changes to the required package.

**Reply by oyejorge** [reply]
To get v4 working, there are a couple differences.

Using the `/public` breaks `npm run build` so use a different folder like `local-dev`

Then, in package.json, change 
> "dev": "vite",

to

> "dev": "vite local-dev",

**Comment by EbrahimHamzeh** [comment]
![image](https://user-images.githubusercontent.com/15343461/189952665-0ad55fcd-2f7c-4b9c-a814-6caeddd7022b.png)
I get a 404 error when I run it.
Sorry to disturb you.

**Reply by vkurko** [reply]
> I get a 404 error when I run it.

Sorry, it's my fault. I don't include the web root directory in the repo. I have updated my previous comment with steps where you should copy the contents of the `docs` directory and use it as your web root. Please let me know if this helps.

**Reply by EbrahimHamzeh** [reply]
![image](https://user-images.githubusercontent.com/15343461/190189579-00909363-d098-4e05-a7ea-9bf15f8ee78b.png)

Hi dear friend
I apologize.
I did what you said. But I didn't succeed and it still gives a 404 error
I could not run it. Can you give me an executable version like your demo so I can edit it?

**Reply by emrahnalci** [reply]
Write folder name in your link

**Reply by vkurko** [reply]
> I did what you said. But I didn't succeed and it still gives a 404 error I could not run it.

You don't have `public` directory with `index.html` in it. Make sure that you copy `docs` into `public` and you create a symbolic link `public/build` which points to `../packages/build`:

```bash
cp -r docs public
ln -s ../packages/build public/build
```

**Reply by vkurko** [reply]
@EbrahimHamzeh 
The directory structure should be like this:
```
calendar/
  docs/
  packages/
  public/
    build -> ../packages/build/
    index.html
  scripts/
```

To create a symbolic link on Windows, see the `mklink` [documentation](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/mklink).

**Reply by EbrahimHamzeh** [reply]
Thank you very much friends @vkurko  @emrahnalci 
My problem with creating the public folder and copying the contents of the doc inside it. The problem was solved and I was able to run the project
Thankful
